gadget/kcmdline.go

```diff
@@ -0,0 +1,62 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package gadget
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type kernelArgsSet map[osutil.KernelArgument]struct{}
+
+// CheckCmdlineAllowed returns an error if an argument from cmdline is
+// not on a list of allowed kernel arguments. A wild card ('*') can be
+// used in the allow list for the values.
+func CheckCmdlineAllowed(cmdline string, allowedLs []string) error {
+	// Set of allowed arguments
+	allowed := kernelArgsSet{}
+	for _, p := range allowedLs {
+		parsed := osutil.KernelParseCommandline(p)
+		if len(parsed) != 1 {
+			return fmt.Errorf("internal error: %q should contain only one kernel argument", p)
+		}
+		allowed[parsed[0]] = struct{}{}
+	}
+
+	proposed := osutil.KernelParseCommandline(cmdline)
+
+	for _, p := range proposed {
+		if allowed.contains(p) ||
+			allowed.contains(osutil.KernelArgument{Param: p.Param, Value: "*"}) {
+			continue
+		}
+
+		return fmt.Errorf("\"%s=%s\" is not an allowed kernel argument",
+			p.Param, p.Value)
+	}
+
+	return nil
+}
+
+func (m kernelArgsSet) contains(arg osutil.KernelArgument) bool {
```

Comment:
(niptick^2) Is there a downside of just using ` map[osutil.KernelArgument]bool` and not using `contains`? Something like:
```go
diff --git a/gadget/kcmdline.go b/gadget/kcmdline.go
index 33ef7f31c6..31abfb6889 100644
--- a/gadget/kcmdline.go
+++ b/gadget/kcmdline.go
@@ -25,7 +25,7 @@ import (
        "github.com/snapcore/snapd/osutil"
 )
 
-type kernelArgsSet map[osutil.KernelArgument]struct{}
+type kernelArgsSet map[osutil.KernelArgument]bool
 
 // CheckCmdlineAllowed returns an error if an argument from cmdline is
 // not on a list of allowed kernel arguments. A wild card ('*') can be
@@ -38,14 +38,13 @@ func CheckCmdlineAllowed(cmdline string, allowedLs []string) error {
                if len(parsed) != 1 {
                        return fmt.Errorf("internal error: %q should contain only one kernel argument", p)
                }
-               allowed[parsed[0]] = struct{}{}
+               allowed[parsed[0]] = true
        }
 
        proposed := osutil.KernelParseCommandline(cmdline)
 
        for _, p := range proposed {
-               if allowed.contains(p) ||
-                       allowed.contains(osutil.KernelArgument{Param: p.Param, Value: "*"}) {
+               if allowed[p] || allowed[osutil.KernelArgument{Param: p.Param, Value: "*"}] {
                        continue
                }
 
@@ -55,8 +54,3 @@ func CheckCmdlineAllowed(cmdline string, allowedLs []string) error {
 
        return nil
 }
-
-func (m kernelArgsSet) contains(arg osutil.KernelArgument) bool {
-       _, ok := m[arg]
-       return ok
-}
```
I know that in go struct{}{} is slightly more efficient but I find the bool one slightly easier to read?